scripts/e2e-test.sh

```diff
@@ -1,15 +1,22 @@
 #!/bin/bash
+set -e # bail on errors
```

Comment:
I dropped chaining commands with `&&` in favour of this so the script fails when any command fails (with exception to `if` statement conditions in this case)

check https://www.gnu.org/software/bash/manual/html_node/The-Set-Builtin.html for `set` command (and `-e` flag in particular) explanation

Reply:
Just to showcase that this continue to fail tests properly - https://app.circleci.com/pipelines/github/gatsbyjs/gatsby/50132/workflows/a257b5b7-1e8c-4de1-9854-77c7e530725b is the intentionally not passing jest test ( from temporary draft PR just to test that - https://github.com/gatsbyjs/gatsby/pull/27182 )

Reply:
(and reason for dropping chaining is that becomes unreadable with conditional commands - like the chmod that is only done if the file actually exists that I added in this PR)